lms/djangoapps/course_home_api/progress/v1/serializers.py

```diff
@@ -1,9 +1,62 @@
 """
 Progress Tab Serializers
 """
-
+import uuid
 from rest_framework import serializers
 from lms.djangoapps.course_home_api.outline.v1.serializers import CourseBlockSerializer
+from rest_framework.reverse import reverse
+
+
+class CoursewareSummarySerializer(serializers.Serializer):
+    """
+    Serializer
+    """
+    blocks = serializers.SerializerMethodField()
+
+    def get_blocks(self, blocks):
+        return [
+            {
+                'display_name': chapter['display_name'],
+                'chapter_url': reverse(  # currently points to the LMS url not the MFE
+                    'jump_to',
+                    kwargs={'course_id': str(self.context['course_key']), 'location': str(chapter['url_name'])},
+                    request=self.context['request'],
+                ),
+                'sections': [
+                    {
+                        'display_name': str(section.display_name),
+                        'section_url': reverse(
+                            'jump_to',
+                            kwargs={'course_id': str(self.context['course_key']), 'location': str(section.url_name)},
+                            request=self.context['request'],
+                        ),
+                        'percent_graded': section.percent_graded,
+                        'graded_total': {
+                            'first_attempted': section.graded_total.first_attempted,
+                            'possible': section.graded_total.possible,
+                            'graded': section.graded_total.graded,
+                            'earned': section.graded_total.earned,
+                        },
+                        'format': section.format,
+                        'due': section.due,
+                        'override': section.override,  # this may not actually be serializable
```

Comment:
Perhaps not in this format, but you can always pass in a sub-serializer or something as appropriate.
We may actually want to do something like that for the chapters and sections here for improved readability. Let's talk about it and maybe @mikix has thoughts.

Reply:
Yah, I agree that it would probably help readability. Might not be used in other contexts, so not likely to help with code reuse, but maybe down the road.